components/ImageGallery.js

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react';
+import { Pressable } from 'react-native';
+import { ReactNativeZoomableView } from '@openspacelabs/react-native-zoomable-view';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+import { HorizontalPager } from './HorizontalPager';
+import { Image } from './Image';
+import { LoadingIndicator } from './LoadingIndicator';
+import { View } from './View';
+
+const IMAGE_PREVIEW_MODE = 'imagePreview';
+const IMAGE_GALLERY_MODE = 'gallery';
+
+function ImageGallery({
+  data,
+  onIndexSelected,
+  onModeChanged,
+  renderOverlay,
+  renderImageOverlay,
+  renderPlaceholder,
+  selectedIndex,
+  showNextPage,
+  style,
+}) {
+  const [currentIndex, setCurrentIndex] = useState(selectedIndex);
+  const [mode, setMode] = useState(IMAGE_GALLERY_MODE);
+
+  function onImageTap(newMode) {
+    if (mode === IMAGE_PREVIEW_MODE) {
+      setMode(IMAGE_GALLERY_MODE);
+    } else {
+      setMode(IMAGE_PREVIEW_MODE);
+    }
+
+    if (_.onModeChanged(newMode)) {
+      onModeChanged(newMode);
+    }
+  }
+
+  function handleIndexSelected(newIndex) {
+    setCurrentIndex(newIndex);
+
+    if (_.isFunction(onIndexSelected)) {
+      onIndexSelected(newIndex);
+    }
+  }
+
+  function renderImage(imageProps) {
+    const imageStyle = imageProps?.style;
+    const resolvedImageProps = _.omit(imageProps, 'style');
+
+    return (
+      <ReactNativeZoomableView
+        maxZoom={1.5}
+        minZoom={0.5}
+        zoomStep={0.5}
+        initialZoom={1}
+        bindToBorders
+      >
+        <Pressable onPress={onImageTap}>
+          <Image style={[imageStyle, style.image]} {...resolvedImageProps} />
```

Comment:
let's memo the style

Reply:
I don't see an easy way to do this since `imageStyle` is a prop from `renderPage` and it changes for each page